### PR TITLE
Feature/add error handling

### PR DESCRIPTION
**Pull request by tlcraft**
I added default values to the initial API endpoint parameters and added a try/except to wrap the workflow.

### CODE REVIEW

File: src/api/index.py [hunk 1]
```diff
@@ -36,10 +36,13 @@ async def root():
 
 @app.post("/analyze_video")
 async def analyze_video(
-    video_url: str, target_item: str
+    video_url: str = "https://youtu.be/MNn9qKG2UFI", target_item: str = "car"
```
Comment (vcalderon2009) [inline src/api/index.py]: I like this. I would instead import the `default_variables` module and then use the `video_url` as the value for `video_url`. Something like this:

```python
from src.utils import default_variables as dv
from typing import Optional

...
video_url: Optional[str]=dv.video_url, target_item: Optional[str]=dv.target_item
```

, while also adding the `target_item` to `default_variables`.

**Review by vcalderon2009 — CHANGES_REQUESTED** [review]
I like that the code is now able to handler errors for this. I added a request for a minor change, but it looks good overall. Thanks for adding this.

File: src/api/index.py (hunk 1 above)
Comment (tlcraft) [reply src/api/index.py]: I like that plan!

Comment (vcalderon2009) [reply src/api/index.py]: Thanks for updating the code!

**Review by vcalderon2009 — APPROVED** [review]

**Pull request merged**